numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -3646,55 +3629,40 @@ dragon4_scientific(PyObject *NPY_UNUSED(dummy), PyObject *args, PyObject *kwds)
+
     if (unique == 0 && precision < 0) {
         PyErr_SetString(PyExc_TypeError,
             "in non-unique mode `precision` must be supplied");
         return NULL;
     }
 
-    return Dragon4_Scientific(obj, unique, precision, sign,
-                              trim, pad_left, exp_digits);
+    return Dragon4_Scientific(obj, digit_mode, precision, sign, trim,
+                              pad_left, exp_digits);
 }
 
 /*
  * Prints floating-point scalars usign the Dragon4 algorithm, positional mode.
- * Arguments:
- *  x - a numpy scalar of Floating type
- *  precision - number of fractional digits to show. In unique mode, can be
- *              ommited and the unique repr will be returned, otherwise the
- *              unique value will be truncated to this number of digits
- *              (breaking the uniqueness guarantee). In fixed mode, is
- *              required, and specifies the number of fractional digits to
- *              print.
- *  unique - whether to use unique (default) or fixed mode.
- *  sign - whether to show the sign for positive values. Default False
- *  trim - one of 'k', '.', '0', '-' to control trailing digits, as follows:
- *          k : don't trim zeros, always leave a decimal point
- *          . : trim all but the zero before the decimal point
- *          0 : trim all trailing zeros, leave decimal point
- *          - : trim trailing zeros and a trailing decimal point
- *         Default is k.
- *  pad_left - pads left side of string with whitespace until at least
- *             this many characters are to the left of the decimal point. If
- *             -1, don't add any padding. Default -1.
- *  pad_right - pads right side of string with whitespace until at least
- *             this many characters are to the right of the decimal point. If
- *             -1, don't add any padding. Default -1.
+ * See docstring of `np.format_float_positional` for description of arguments.
+ * The differences is that a value of -1 is valid for pad_left, pad_right,
+ * precision, which is equivalent to `None`.
  */
 static PyObject *
 dragon4_positional(PyObject *NPY_UNUSED(dummy), PyObject *args, PyObject *kwds)
 {
     PyObject *obj;
-    static char *kwlist[] = {"x", "precision", "unique", "sign", "trim",
-                             "pad_left", "pad_right", NULL};
+    static char *kwlist[] = {"x", "precision", "unique", "fractional",
+                             "sign", "trim", "pad_left", "pad_right", NULL};
     int precision=-1, pad_left=-1, pad_right=-1;
     char *trimstr=NULL;
+    CutoffMode cutoff_mode;
+    DigitMode digit_mode;
     TrimMode trim = TrimMode_None;
-    int sign=0, unique=1;
+    int sign=0, unique=1, fractional=0;
 
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|iiisii", kwlist,
-                &obj, &precision, &unique, &sign, &trimstr, &pad_left,
-                &pad_right)) {
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|iiiisii", kwlist,
```

Comment:
This should have `:dragon4_position` after the list of types, to produce a more useful error.